railties/lib/rails/engine/commands.rb

```diff
@@ -19,6 +23,20 @@
   engine.load_generators
   require "rails/commands/#{command}"
 
+when 'console', 'server', 'dbconsole', 'runner', 'version', 'help'
+  require "rails/commands/commands_tasks"
+
+  while Dir.pwd != "/"
```

Comment:
Good call. I just switched it to `Pathname.pwd.root?`, which says it has 'experimental' support for non-Unix platforms. Rails.root is not net unfortunately.